python/ray/autoscaler/autoscaler.py

```diff
@@ -158,6 +158,13 @@ def __init__(self):
 
     def update(self, ip, static_resources, dynamic_resources):
         self.static_resources_by_ip[ip] = static_resources
+        # TODO(romilb): This is a hotfix to make autoscaler work with #4555
+        dynamic_resources_update = dynamic_resources.copy()
+        for resource_name, capacity in static_resources.items():
+            if resource_name not in dynamic_resources_update:
+                dynamic_resources_update[resource_name] = 0.0
+        self.dynamic_resources_by_ip[ip] = dynamic_resources_update
+
         self.dynamic_resources_by_ip[ip] = dynamic_resources
```

Comment:
Line 168 and Line 166 do not make sense together, should 168 be removed?

Reply:
Yes I think so, thanks!